netsync/manager.go

```diff
@@ -175,6 +176,14 @@ type SyncManager struct {
 	feeEstimator *mempool.FeeEstimator
 }
 
+// SyncPeer returns the currrent syncPeer in a read mutex.
+func (sm *SyncManager) SyncPeer() *peerpkg.Peer {
+	sm.syncPeerMutex.RLock()
+	defer sm.syncPeerMutex.RUnlock()
+
+	return sm.syncPeer
+}
+
```

Comment:
I must have a misunderstanding. This locks and unlocks the mutex, effectively doing nothing before returning the syncPeer? Could you tell me if this is an idiom I don't know?

I checked neutrino and it's exactly the same (except not an RWMutex).

Reply:
Maybe they intended to also return an unlocker function that could be used by the caller instead of touching the internal mutex...?

Reply:
You just explained this to me. Sorry I didn't see it. It is protecting syncPeer itself.